help_to_heat/frontdoor/views.py

```diff
@@ -282,6 +286,75 @@ class OwnPropertyView(PageView):
     def get_context(self, *args, **kwargs):
         return {"own_property_options_map": schemas.own_property_options_map}
 
+    def handle_post(self, request, session_id, page_name, data, is_change_page):
+        prev_page_name, next_page_name = get_prev_next_page_name(page_name)
+        data = request.POST.dict()
+        own_property = data.get("own_property")
+
+        if own_property == "Yes, I own my property and live in it":
+            next_page_name = "park-home"
+
+        if is_change_page:
+            assert page_name in schemas.change_page_lookup
+            next_page_name = schemas.change_page_lookup[page_name]
+
+        return redirect("frontdoor:page", session_id=session_id, page_name=next_page_name)
+
+
+@register_page("park-home")
+class ParkHomeView(PageView):
+    def get_context(self, *args, **kwargs):
+        return {"park_home_options_map": schemas.park_home_options_map}
+
+    def handle_post(self, request, session_id, page_name, data, is_change_page):
+        prev_page_name, next_page_name = get_prev_next_page_name(page_name)
+        data = request.POST.dict()
+        park_home = data.get("park_home")
+
+        if park_home == "Yes":
+            next_page_name = "park-home-main-residence"
+
+        if is_change_page:
+            if park_home == "Yes":
+                return redirect("frontdoor:change-page", session_id=session_id, page_name=next_page_name)
+            else:
+                assert page_name in schemas.change_page_lookup
+                next_page_name = schemas.change_page_lookup[page_name]
+
+        return redirect("frontdoor:page", session_id=session_id, page_name=next_page_name)
+
+
+@register_page("park-home-main-residence")
+class ParkHomeMainResidenceView(PageView):
+    def get_context(self, *args, **kwargs):
+        return {"park_home_main_residence_options_map": schemas.park_home_main_residence_options_map}
+
+    def handle_post(self, request, session_id, page_name, data, is_change_page):
+        prev_page_name, next_page_name = get_prev_next_page_name(page_name)
+        data = request.POST.dict()
+        park_home_main_residence = data.get("park_home_main_residence")
+
+        if park_home_main_residence == "No":
+            next_page_name = "park-home-application-closed"
+
+        if is_change_page:
+            if park_home_main_residence == "No":
+                return redirect("frontdoor:page", session_id=session_id, page_name=next_page_name)
+            else:
+                assert page_name in schemas.change_page_lookup
+                next_page_name = schemas.change_page_lookup[page_name]
```

Comment:
This looks weird to me. Why do we have the first `if`? It could be replaced by putting the `next_page_name` assignment in the other `if park_home_main_residence == "No"` block or having the return say `page_name="park-home-application-closed"`.

Reply:
Or we could even just have
```python
if park_home_main_residence == "No":
    next_page_name = "park-home-application-closed"
elif is_change_page:
    assert page_name in schemas.change_page_lookup
    next_page_name = schemas.change_page_lookup[page_name]

return redirect("frontfoor:page", session_id=session_id, page_name=next_page_name)
```

Reply:
Thanks for pointing out - I figured out we actually don't need `is_change_page` in this two classes in this branch 😅